### PR TITLE
Fix Electron build

### LINKED ISSUE #1273: Get Electron build working
Since becoming maintainer of this project, and making some security updates, we have not been able to get the electron build working. This is why there has not been a release in a while.

This is no small undertaking, but very very important!

### PR DESCRIPTION
**Pull request by vardhanapoorv**
Fixes #1273 .

Changes proposed in this pull request:

- Upgrade `electron-builder`, `electron-devtools-installer` & `webpack-dev-server` packages. 

- Fix Babel config.

### CODE REVIEW

**Comment by acao** [comment]
thanks so much for working on this @vardhanapoorv ! hoping to finally check this out this weekend and give it a local run through

**Comment by vardhanapoorv** [comment]
That's great!

**Comment by acao** [comment]
wasn't able to get it running last weekend, hopefully I can get to it tonight!

**Comment by vardhanapoorv** [comment]
@acao Let me know if anymore changes are needed, when you get a chance to review it.

**Comment by acao** [comment]
@vardhanapoorv nice! it works in terms of fixing the initial schema choice screen

on `yarn start`, once I supply a remote endpoint though, i get a blank screen and this error
image

**Comment by vardhanapoorv** [comment]
@acao I build it using `npm run build` and then tested release app with `npm run release:mac` before.
.

**Comment by vardhanapoorv** [comment]
@acao just tested with `yarn start` that also seems to be working for me. Let me know if I'm doing something wrong here.
Screenshot 2021-04-04 at 11 35 55 AM

**Comment by acao** [comment]
so, when I run `yarn` after cloning your PR, the yarn.lock file is very different. so this might be why its not working here or in CI. If you're able to submit a PR with a clean lockfile then we can move forward! by that i mean, when you run `yarn` locally there should be no change to yarn.lock.

to try and simulate the situation I'm having, try cloning this repo into a new folder, and running yarn fresh, and trying the above commands, and you'll see the lockfile diff and other issues I'm talking about

**Comment by vardhanapoorv** [comment]
@acao oh yeah right makes sense, thanks. This reminds me that I was facing some issue with global lockfile so I installed the dependencies for electron package from there which created a separate lockfile in that folder, which I'm thinking is wrong since no file existed before. I'm a bit confused with the build process of the packages in this repo.
After your above comment, I did try a bit but I'm not completely clear if the process I'm following is correct or not.  
Can you tell me the steps which should be followed? Should I install dependencies from base folder or from the respective packages folder? Then should I build the packages separately from their respective folders(React build worked, but the html build failed for me)?
Thanks for helping out.

**Comment by vardhanapoorv** [comment]
@acao ignore the above questions, I actually managed to update it correctly I think. Take a look and let me know if you still face any errors.

**Comment by acao** [comment]
awesome! also, can you sign the CLA for GraphQL Foundation? this is a new thing now that playground is a GraphQL Foundation project, instead of Prisma's CLA bot.

**Comment by vardhanapoorv** [comment]
Done!

**Comment by acao** [comment]
@vardhanapoorv it works great for me locally! excellent work. not easy to deal with all these package resolutions!

File: packages/graphql-playground-electron/webpack.config.build.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ if (!fs.existsSync(appEntrypoint)) {
 
 module.exports = {
   devtool: 'source-map',
+  mode: 'production',
```
Comment (acao) [inline packages/graphql-playground-electron/webpack.config.build.js:21]: :1st_place_medal:

File: packages/graphql-playground-electron/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "removeComments": true,
     "module": "commonjs",
-    "jsx": "preserve",
+    "jsx": "react",
```
Comment (acao) [inline packages/graphql-playground-electron/tsconfig.json:5]: :1st_place_medal:

**Review by acao — APPROVED** [review]

**Comment by vardhanapoorv** [comment]
> @vardhanapoorv it works great for me locally! excellent work. not easy to deal with all these package resolutions!

That's great, thanks for helping me out with my questions throughout this issue!

**Comment by acao** [comment]
@vardhanapoorv glad to help! the contribution effort is very welcome! if you'd like to help accelerate the follow up, here's what I had in mind for next:

- upgrade codemirror-graphql
- upgrade electron
- add electron build to Github Releases using atlassian changesets and github actions

**Comment by vardhanapoorv** [comment]
Absolutely, sounds good. I can tackle these next. As a separate PR right?

**Comment by acao** [comment]
@vardhanapoorv yes! separate PR for sure. I will merge this one now. Again, thanks so much!

**Pull request merged**